src/lib/hooks/use-balance.ts

```diff
@@ -77,7 +77,7 @@ export function useBalances(address?: string, tokens?: string[]) {
   })
 
   const fetchBalances = useCallback(async () => {
-    if (!address || !publicClient || !tokens || tokens.length === 0) return
+    if (!address || !publicClient || !tokens || tokens.length === 0) return []
```

Comment:
This is basically happening in `select` of the query. So could be removed there.